File: python/program/ParkingLot.py
class ParkingLot:
    def __init__(self, kapasitas):
        self._kapasitas = kapasitas
        self._vehicles = []

    def get_kapasitas(self):
        return self._kapasitas

    def set_kapasitas(self, kapasitas):
        self._kapasitas = kapasitas

    def parkir_kendaraan(self, vehicle):
        if len(self._vehicles) < self._kapasitas:
            self._vehicles.append(vehicle)
            print("Parkir berhasil, total kendaraan sekarang:", len(self._vehicles))
        else:
            print("Parkir penuh.")

    def is_full(self):
        return len(self._vehicles) >= self._kapasitas

    def is_empty(self):
        return len(self._vehicles) == 0

File: cpp/program/ParkingLot.cpp
#include <iostream>
#include <vector>
#include "Vehicle.h"

using namespace std;

class ParkingLot {
private:
    int kapasitas;
    vector<Vehicle*> vehicles;

public:
    ParkingLot(int kapasitas) : kapasitas(kapasitas) {}

    int getKapasitas() const {
        return kapasitas;
    }

    vector<Vehicle*> getVehicles() const {
        return vehicles;
    }

    void setKapasitas(int newKapasitas) {
        kapasitas = newKapasitas;
    }

    void parkirKendaraan(Vehicle* vehicle) {
        if (vehicles.size() < kapasitas) {
            vehicles.push_back(vehicle);
            cout << "Parkir berhasil, total kendaraan sekarang: " << vehicles.size() << endl;
        } else {
            cout << "Parkir penuh." << endl;
        }
    }

    bool isFull() const {
        return vehicles.size() >= kapasitas;
    }

    bool isEmpty() const {
        return vehicles.empty();
    }
};
